spec/registry.md

```diff
@@ -255,3 +255,500 @@ which only expects the `accord` option:
+> ```
+> .local $example = {|-1234.567| :number}
+> {{{$num :number} == {$example}}}
+> ```
+
+> [!NOTE]
+> Implementations are encouraged to provide support for compound types or data structures
+> that provide additional semantic meaning to the formatting of number-like values.
+> For example, in ICU4J, the type `com.ibm.icu.util.Measure` can be used to communicate
+> a value that include a unit
+> or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
+> options (such as the number of fraction digits).
+
+
+### Options
+
+The following options and their values are required in the default registry to be available on the 
+function `:number`:
+- `select`
+   -  `plural` (default)
+   -  `ordinal`
+   -  `exact`
+
+ 
+> [!IMPORTANT]
+> **Default Value of `select` Option**
+>
+> The value `plural` is default for the option `select` 
+> because it is the most common use case for numeric selection.
+> It can be used for exact value matches but also allows for the grammatical needs of other 
+> languages using CLDR's plural rules.
+> This might not be noticeable in the source language (particularly English), 
+> but can cause problems in target locales that the original developer is not considering.
+>
+> For example, a naive developer might use a special message for the value `1` without
+> considering other locale's need for a `one` plural:
+>
+>```
+> .match {$var}
+> 1   {{You have one last chance}}
+> one {{You have {$var} chance remaining}} // needed by languages such as Polish or Russian
+>                                          // such locales typically require other keywords
+>                                          // such as two, few, many, and so forth
+> *   {{You have {$var} chances remaining}}
+>```
+
+- `compactDisplay` (this option only has meaning when combined with the option `notation=compact`)
+   - `short` (default)
+   - `long`
+- `notation`
+   - `standard` (default)
+   - `scientific`
+   - `engineering`
+   - `compact`
+- `numberingSystem`
+   - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+     (default is locale-specific)
+- `signDisplay`
+   -  `auto` (default)
+   -  `always`
+   -  `exceptZero`
+   -  `negative`
+   -  `never`
+- `style`
+  - `decimal` (default)
+  - `percent` (see [Percent Style](#percent-style) below)
+- `useGrouping`
+  - `auto` (default)
+  - `always`
+  - `never`
+  - `min2`
+- `minimumIntegerDigits`
+  - (non-negative integer, default: `1`)
+> [!NOTE]
+> The following options do not have default values because they are only to be used
+> as overrides for an existing locale-and-value dependent implementation-defined
+> default
+
+- `minimumFractionDigits`
+  - (non-negative integer)
+- `maximumFractionDigits`
+  - (non-negative integer)
+- `minimumSignificantDigits`
+  - (non-negative integer)
+- `maximumSignificantDigits`
+  - (non-negative integer)
```

Comment:
I'm not convinced that this note is helpful; the same dynamic default behavior is also true of `numberingSystem`.
```suggestion
- `minimumFractionDigits`
  - (non-negative integer) (default is implementation-dependent and may depend upon locale and/or value and/or other options)
- `maximumFractionDigits`
  - (non-negative integer) (default is implementation-dependent and may depend upon locale and/or value and/or other options)
- `minimumSignificantDigits`
  - (non-negative integer) (default is implementation-dependent and may depend upon locale and/or value and/or other options)
- `maximumSignificantDigits`
  - (non-negative integer) (default is implementation-dependent and may depend upon locale and/or value and/or other options)
```

Reply:
Could we instead put something above all of the options in a non-NOTE style that says something like:

> Some options do not have default values defined in this specification.
> The defaults for these options are implementation-dependent.
> In general, the defaults depend on the locale, the value of other options,
> or both.

Reply:
I implemented this. See what you think.